src/test/java/uk/gov/hmcts/reform/sscs/service/LetterUtilsTest.java

```diff
@@ -415,4 +410,260 @@ private String getExpectedName(final String otherPartyId, final NotificationWrap
+                .build();
+
+        Representative representative = Representative.builder()
+            .id("REP123456")
+            .name(Name.builder()
+                .title("Mr.")
+                .firstName("Representative")
+                .lastName("Appellant")
+                .build())
+            .build();
+
+        caseData = SscsCaseData.builder()
+                .jointParty(jointPartyWithId)
+                .otherParties(buildOtherPartyData())
+                .appeal(Appeal.builder()
+                    .appellant(appellantWithId)
+                    .rep(representative)
+                    .build())
+                .build();
+    }
+
+    private List<CcdValue<OtherParty>> buildOtherPartyData() {
+        return List.of(CcdValue.<OtherParty>builder()
+            .value(OtherParty.builder()
+                .id("OP123456")
+                .name(Name.builder()
+                    .firstName("Other")
+                    .lastName("Party")
+                    .build())
+                .otherPartySubscription(Subscription.builder().email("other@party").subscribeEmail("Yes").build())
+                .rep(Representative.builder()
+                    .id("OPREP123456")
+                    .name(Name.builder()
+                        .firstName("OtherParty")
+                        .lastName("Representative")
+                        .build())
+                    .hasRepresentative(YesNo.YES.getValue())
+                    .build())
+                .build())
+            .build(),
+            CcdValue.<OtherParty>builder()
+                .value(OtherParty.builder()
+                    .id("OP7890123")
+                    .name(Name.builder()
+                        .firstName("Other 1")
+                        .lastName("Party 1")
+                        .build())
+                    .otherPartySubscription(Subscription.builder().email("other@party").subscribeEmail("Yes").build())
+                    .rep(Representative.builder()
+                        .hasRepresentative(YesNo.NO.getValue())
+                        .build())
+                    .build())
+                .build()
+            );
+    }
+
+    @DisplayName("When sender is appellant, representative or Joint party"
+            + " then return name of corresponding party")
+    @Test
+    @Parameters({"appellant,Tom Cat","representative,Representative Appellant", "jointParty,Joint Party", "jointParty1, "})
+    public void testGetNameForSenderRepresentative(String senderType, String senderName) {
+        DynamicList sender = new DynamicList(new DynamicListItem(senderType, senderType), new ArrayList<>());
+        caseData.setOriginalSender(sender);
+        assertEquals(senderName, LetterUtils.getNameForSender(caseData));
+    }
+
+    @DisplayName("When sender is null then return name of case Appellant")
```

Comment:
This display name doesn't match up with the method name or contents

Reply:
changed display name to match up test